libs/sdk-ui-kit/src/List/Listbox.tsx

```diff
@@ -0,0 +1,238 @@
+
+    onSelect?: (item: IListboxItem<T>) => void;
+    onClose?: () => void;
+
+    selectedItemId?: string;
+
+    ItemComponent?: React.ComponentType<IListboxItemProps<any>>;
+
+    ariaAttributes: IDropdownBodyRenderProps["ariaAttributes"];
+}
+
+const makeItemId = (listboxId: string, item: IListboxItem<unknown>) => `item-${listboxId}-${item.id}`;
+
+/**
+ * An accessible listbox component that can be navigated by keyboard.
+ * Usable in a <Dropdown /> component.
+ * Should implement https://www.w3.org/WAI/ARIA/apg/patterns/listbox/
+ * @internal
+ */
+export function Listbox<T>({
+    items,
+    className,
+    maxWidth,
+    onSelect,
+    onClose,
+    selectedItemId,
+    ariaAttributes,
+    ItemComponent = DefaultListboxItemComponent,
+}: IListboxProps<T>): React.ReactNode {
+    const [focusedIndex, setFocusedIndex] = React.useState<number>(() => {
+        const selectedIndex = items.findIndex((item) => item.id === selectedItemId);
+        return selectedIndex >= 0 ? selectedIndex : 0;
+    });
+
+    const itemRefs = React.useRef<(HTMLLIElement | null)[]>([]);
+
+    // Update refs array size when items change. The actual refs are updated during render.
+    React.useEffect(() => {
+        itemRefs.current = itemRefs.current.slice(0, items.length);
+    }, [items]);
+
+    // Scroll focused item into view
+    React.useEffect(() => {
+        const focusedItem = itemRefs.current[focusedIndex];
+        if (focusedItem) {
+            focusedItem.scrollIntoView({ block: "nearest" });
+        }
+    }, [focusedIndex]);
+
+    const handleCharacterSearch = React.useCallback(
+        (char: string) => {
+            if (char.length !== 1 || !/[a-z0-9]/.test(char)) {
+                return;
+            }
+
+            const itemIndex = items.findIndex(
+                (item, index) =>
+                    index > focusedIndex &&
+                    item.stringTitle.toLowerCase().startsWith(char) &&
+                    !item.isDisabled,
+            );
+
+            if (itemIndex !== -1) {
+                setFocusedIndex(itemIndex);
+                return;
+            }
+
+            // If not found after the current index, start from the beginning
+            const fromStartIndex = items.findIndex(
+                (item) => item.stringTitle.toLowerCase().startsWith(char) && !item.isDisabled,
+            );
+
+            if (fromStartIndex !== -1) {
+                setFocusedIndex(fromStartIndex);
+            }
+        },
+        [focusedIndex, items],
+    );
+
+    const handleSelectItem = React.useCallback(
+        (item?: IListboxItem<T>) => {
+            if (!item || item.isDisabled) {
+                return;
+            }
+            onSelect?.(item);
+            onClose?.();
+        },
+        [onClose, onSelect],
+    );
+
+    const handleKeyDown = React.useCallback(
+        (event: React.KeyboardEvent<HTMLUListElement>) => {
+            switch (event.code) {
+                case "ArrowDown":
```

Comment:
this keyboard handling is quite similar to this
https://github.com/gooddata/gooddata-ui-sdk/pull/6258/files#diff-a1eedcef130bcb5ddb4733e5659888870e8d69d0dce3ab6306c5092c0348a968R99
pls sync with @NestorEncinas if we can have single list supporting keyboard navigation